### ISSUE TITLE
[Source Control] Changes not showing

### ISSUE THREAD

**Issue by RaniAgus** [microsoft/vscode#201717]
As you can see in the screenshot below, I have changes in three files, but none of them appear un source control tab:

![image](https://github.com/microsoft/vscode/assets/39303639/e4883072-b828-4bd7-a450-9c251163a454)

Does this issue occur when all extensions are disabled?: Yes

- VS Code Version: 
- OS Version: Windows 11 Pro - 22H2 - 22621.2861

Steps to Reproduce:

1. Make changes in source code
2. Open "Source Control" tab

Also, sometimes "Refresh" button is missing, but I wasn't able to find a proper way to reproduce this, sorry 😞

Please let me know if you need more info about the bug. Thanks.

**Comment by lszomoru** [comment]
@RaniAgus, are you able to reproduce this issue with the latest VS Code Insiders release?

**Comment by TsengTofu** [comment]
same issue here

**Comment by koyuboy** [comment]
I have the same problem. **When I cloned the repository to a different location, the problem was fixed for a while, but then it broke again.**

```
Version: 1.85.1 (user setup)
Commit: 0ee08df0cf4527e40edc9aa28f4b5bd38bbff2b2
Date: 2023-12-13T09:49:37.021Z
Electron: 25.9.7
ElectronBuildId: 25551756
Chromium: 114.0.5735.289
Node.js: 18.15.0
V8: 11.4.183.29-electron.0
OS: Windows_NT x64 10.0.22621
```

**Comment by lszomoru** [comment]
@RaniAgus, @TsengTofu, @koyuboy, are you able to reproduce if you set the following settings to `never`:
* scm.showIncomingChanges
* scm.showOutgoingChanges 

Also, could you please try the latest VS Code Insiders release and let me know if you are still able to reproduce the issue? Thanks!

**Comment by TofuTseng** [comment]
@lszomoru 
I've set these two to `'Never'` in my Visual Studio Code.
<img width="385" alt="image" src="https://github.com/microsoft/vscode/assets/75705724/b5eb94c7-e5e8-4631-b933-179e4c108051">
<img width="387" alt="image" src="https://github.com/microsoft/vscode/assets/75705724/2e100dd0-0055-4197-af0c-71c36799f51a">

It seems worked for me.
But don't know why and how it works.
And, what is the latest VS Code Insiders? I'm not sure that I know what it means?
Thanks a lot!

**Comment by lszomoru** [comment]
@TofuTseng, thanks for testing the proposed suggestions. When you toggle the two settings listed above, you are disabling the code path that has introduced the regression which is why it solves the problem.

You can also download the latest VS Code Insiders release from [here](https://code.visualstudio.com/insiders/).

**Comment by martinog2** [comment]
Same here:

Version: 1.85.1 (user setup)
Commit: 0ee08df0cf4527e40edc9aa28f4b5bd38bbff2b2
Date: 2023-12-13T09:49:37.021Z
Electron: 25.9.7
ElectronBuildId: 25551756
Chromium: 114.0.5735.289
Node.js: 18.15.0
V8: 11.4.183.29-electron.0
OS: Windows_NT x64 10.0.19045

**Comment by koyuboy** [comment]
@lszomoru
I set these 2 variables' values to "never" and fixed my problem.  
Thanks a lot!

**Comment by lszomoru** [comment]
Are you able to reproduce with the VS Code Insiders [release](https://code.visualstudio.com/insiders/)?

**Comment by AndrewDryga** [comment]
This still happens with the latest stable release, setting options to "never" fixes the problem for me too.

**Comment by lszomoru** [comment]
The fix for this issue will be available in VS Code 1.86 scheduled to release to stable next week.

**Comment by eleanorjboyd** [comment]
verified this as it seems the same as: https://github.com/microsoft/vscode/issues/200387
